### ISSUE TITLE
[BUG] Chroma DB Duplicate ID Error

### ISSUE THREAD

**Issue by ArmanJR** [embedchain/embedchain#64]
This is my code:

```python
import os
os.environ["OPENAI_API_KEY"] = "sk-???"
from embedchain import App
naval_chat_bot = App()
naval_chat_bot.add_local("pdf_file", "docs/masnavi-en.pdf")
print(naval_chat_bot.query("Who is the most powerful man?"))
```

I get `chromadb.errors.DuplicateIDError: Expected IDs to be unique, found duplicates for`. Where is the problem?

P.S: This was my second attempt. The first one with a different pdf document was successful.

**Comment by luisrock** [comment]
Same issue here

**Comment by missioncfwf** [comment]
Same issue here. I had no problem uploading small files, but big ones cause this issue for me

**Comment by ramaschneider** [comment]
Same here - is there any update regarding this issue?

**Comment by taranjeet** [comment]
I think this could be because ids are clashing.
Right now, the ids are generated using a hash of chunk text and url. There may be scenarios where same chunk is occurring multiple times in a doc, hence the error.

@ramaschneider / @missioncfwf / @ArmanJR : possible for anyone of you to share doc file?
I just want to double check if this is the issue or something else.

**Comment by ArmanJR** [comment]
@taranjeet sure: [https://www.sufi.ir/books/download/english/molavi-en/masnavi-en.pdf](https://www.sufi.ir/books/download/english/molavi-en/masnavi-en.pdf)

**Comment by Harin329** [comment]
I opened a PR for this, setting a random bit in the ID hashing function seems like a harmless and quick fix. Not entirely sure of the implications of turning the lists into a set, but that may affect the composition of the dataset

**Comment by taranjeet** [comment]
Hey @ArmanJR @luisrock @missioncfwf @ramaschneider @Harin329 : merged PR https://github.com/embedchain/embedchain/pull/160 which addresses this issue.

**Comment by taranjeet** [comment]
Released in version 0.0.18
https://test.pypi.org/project/embedchain/0.0.18/

Closing this issue.
